src/lib/components/forms/form/form-components/number-input/number-input.component.spec.ts

```diff
@@ -0,0 +1,143 @@
+    let component: NumberInputComponent;
+    let loader: HarnessLoader;
+    let input: MatInputHarness;
+
+    beforeEach(async () => {
+        TestBed.configureTestingModule({
+            imports: [MatFormFieldModule, MatInputModule, NoopAnimationsModule, FormsModule],
+            declarations: [NumberInputComponent, MockTooltipDirective]
+        });
+
+        fixture = TestBed.createComponent(NumberInputComponent);
+        component = fixture.componentInstance;
+        loader = TestbedHarnessEnvironment.loader(fixture);
+
+        fixture.detectChanges();
+
+        input = await loader.getHarness(MatInputHarness);
+    });
+
+    it('should create', async () => {
+        expect(component).toBeTruthy();
+        expect(input).toBeTruthy();
+    });
+
+    it('should have type number', async () => {
+        expect(await input.getType()).toBe('number');
+    });
+
+    it('should set required validation when #isRequired is set', async () => {
+        expect(await input.isRequired()).toBe(false);
+        component.isRequired = true;
+        fixture.detectChanges();
+
+        expect(await input.isRequired()).toBe(true);
+    });
+
+    it('should disable the input when #isDisabled is set', async () => {
+        expect(await input.isDisabled()).toBe(false);
+        component.isDisabled = true;
+        fixture.detectChanges();
+
+        expect(await input.isDisabled()).toBe(true);
+    });
+
+    it('should have #name as label of the input', async () => {
+        const formField: MatFormFieldHarness = await loader.getHarness(MatFormFieldHarness);
+        expect(await formField.getLabel()).toBe('');
+
+        component.name = 'My Label';
+        fixture.detectChanges();
+        expect(await formField.getLabel()).toBe(component.name);
+    });
+
+    describe('with tooltip', () => {
+        let tooltip: MockTooltipDirective;
+        beforeEach(() => {
+            tooltip = fixture.debugElement.query(By.directive(MockTooltipDirective)).injector.get(MockTooltipDirective);
+        });
+
+        it('should apply the tcTooltip directive to the form field with the given #tooltipText', () => {
+            expect(tooltip).toBeTruthy();
+            expect(tooltip.tcTooltip).toBeUndefined();
+
+            component.tooltipText = 'My tooltip';
+            fixture.detectChanges();
+
+            expect(tooltip.tcTooltip).toBe(component.tooltipText);
+        });
+
+        it(`should set the tooltip's placement according to the input #tooltipPlacement`, () => {
+            expect(tooltip.placement).toBeUndefined();
+
+            component.tooltipPlacement = TerraPlacementEnum.BOTTOM;
+            fixture.detectChanges();
+
+            expect(tooltip.placement).toBe(component.tooltipPlacement);
+        });
+    });
+
+    it('should set min value validation according to input #minValue', () => {
+        const inputElement: HTMLInputElement = fixture.debugElement.query(By.css('input')).nativeElement;
```

Comment:
Ich find die Variante hier mit dem Host-Element ( `expect(+(await host.getAttribute('rows'))).toBe(4);` ) zu arbeiten schöner :)

Reply:
Hab das eben schon in deinem PR gesehen. Wenn dann aber `getProperty()` oder?
Ich muss aber auch sagen, dass man dadurch halt ein bisschen die Typesafety verliert.. Denn das ist ja ein generelles `TestElement` wogegen ich hier ein `HTMLInputElement` habe, also weiß welche properties es hat.